gridtools/shared/xgridobj.py

```diff
@@ -0,0 +1,78 @@
+import ctypes as ct
+from dataclasses import dataclass
+from typing import Optional
+import xarray as xr
+from gridtools.shared.gridtools_utils import check_file_is_there
+from gridtools.shared.gridobj import GridObj
+
+@dataclass
+class XGridObj() :
+
+    src_mosaic : Optional[str] = None
+    tgt_mosaic : Optional[str] = None
+    restart_remap_file : Optional[str] = None
+    out_remap_file     : Optional[str] = 'remap.nc'
+    src_grid : Optional[GridObj] = None 
+    tgt_grid : Optional[GridObj] = None 
+    debug    : Optional[bool] = False
+
+    dataset : Optional[xr.Dataset] = None 
+    _dataset_exists = False
+    
+    def __post_init__(self) :
+
+        if self._check_dataset()            : return
```

Comment:
I think one `if` with `or`'s between the function calls might show the logic here a little more clearly, but thats more of a style preference.

Also I think you need two underscores to make something private, if that's whats intended for the `_check_*` functions and `_dataset_exists` variable

Reply:
quick googling, i think double underscores are involved with name mangling

Reply:
Yeah I think you want the name mangling that's what makes it "private" in Python